com.unity.shadergraph/Editor/Drawing/MaterialEditor/ShaderGraphPropertyDrawers.cs

```diff
@@ -0,0 +1,256 @@
+using System;
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.Rendering;
+using Object = UnityEngine.Object;
+
+namespace UnityEditor.ShaderGraph.Drawing
+{
+    internal static class ShaderGraphPropertyDrawers
+    {
+
+        public static void DrawShaderGraphGUI(MaterialEditor materialEditor, MaterialProperty[] properties)
+        {
+            Material m = materialEditor.target as Material;
+            Shader s = m.shader;
+            string path = AssetDatabase.GetAssetPath(s);
+            ShaderGraphMetadata metadata = null;
+            foreach (var obj in AssetDatabase.LoadAllAssetsAtPath(path))
+            {
+                if(obj is ShaderGraphMetadata meta)
+                {
+                    metadata = meta;
+                    break;
+                }
+            }
+            Debug.Assert(metadata != null, "Cannot draw ShaderGraph GUI on a non-ShaderGraph material", materialEditor.target);
```

Comment:
Do we want AssertHelper here? Throwing an unhandled exception isn't the way to go. If this happens to a user, we can probably just ignore it and return nothing, or else maybe add a text box that says, "This shader is not supported by the current ShaderGUI class."